### PR TITLE
Changing kubelet LegacyHostIPs to InternalIPs

### PR DESCRIPTION
**Pull request by krousey**
Part of an effort to remove LegacyHostIPs. #8087 

@bgrant0607 @dchen1107

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1811,7 +1811,7 @@ func (kl *Kubelet) setNodeStatus(node *api.Node) error {
 	} else {
 		addr := net.ParseIP(kl.hostname)
 		if addr != nil {
-			node.Status.Addresses = []api.NodeAddress{{Type: api.NodeLegacyHostIP, Address: addr.String()}}
+			node.Status.Addresses = []api.NodeAddress{{Type: api.NodeInternalIP, Address: addr.String()}}
```
Comment (bgrant0607) [inline pkg/kubelet/kubelet.go]: I believe this comes from --hostname-override. Unfortunately, we have no way of knowing whether that's internal or external. I think internal is a reasonable guess. It's only used in the case where there isn't a cloud provider. A comment here would be useful.

@dchen1107 @cjcullen

**Comment by bgrant0607** [comment]
I don't think we can stop setting LegacyHostIP until we remove the other uses of it.

**Comment by bgrant0607** [comment]
For example:
https://github.com/GoogleCloudPlatform/kubernetes/blob/6e3ff01fa723afa7529dac113e9b654bd000f19a/test/e2e/service.go#L979

**Comment by krousey** [comment]
@bgrant0607 Is there a way to test that this is ok? Or should we just aim to remove all uses at once?

**Comment by krousey** [comment]
I'm going to abandon this PR for now.

**Pull request closed without merging**